app/src/main/java/com/github/onlynotesswent/ui/navigation/NavigationActions.kt

```diff
@@ -97,4 +98,32 @@ open class NavigationActions(
   open fun currentRoute(): String {
     return navController.currentDestination?.route ?: ""
   }
+
+  /**
+   * Pushes an Id to the screen navigation stack (either folder id or user id)
+   *
+   * @param id The Id to push.
+   */
+  open fun pushToScreenNavigationStack(id: String) {
+    screenNavigationStack.push(id)
```

Comment:
Thats why I clear the stack every time we go to a Top Level destination. I feel it is okey not to check since I dont see any user going to look for an huge number of user profiles in a note-taking app without going back to a top level destination